src/state_machine.zig

```diff
@@ -1599,6 +1890,86 @@ pub fn StateMachineType(
             return self.forest.grooves.posted.get(pending_timestamp);
         }
 
+        fn transfer_update_pending_status(
+            self: *StateMachine,
+            timestamp: u64,
+            status: TransferPendingStatus,
+        ) void {
+            assert(timestamp != 0);
+            assert(status != .none and status != .pending);
+
+            self.forest.grooves.transfers.indexes.pending_status.remove(&.{
+                .timestamp = timestamp,
+                .field = @intFromEnum(TransferPendingStatus.pending),
+            });
+            self.forest.grooves.transfers.indexes.pending_status.put(&.{
+                .timestamp = timestamp,
+                .field = @intFromEnum(status),
+            });
```

Comment:
Ah, so we remove-put instead of .update since we don't prefetch the pending_status?

Reply:
Not sure if I understood what you mean ... but `PendingStatus` is an index, not an ObjectTree.
To update an index, you remove and reinsert it (we do the same in `groove.zig`).